packages/react-aria-components/src/Menu.tsx

```diff
@@ -21,13 +21,17 @@ import {Key, LinkDOMProps} from '@react-types/shared';
 import {KeyboardContext} from './Keyboard';
 import {OverlayTriggerStateContext} from './Dialog';
 import {PopoverContext} from './Popover';
-import {PressResponder} from '@react-aria/interactions';
-import React, {createContext, ForwardedRef, forwardRef, ReactNode, RefObject, useCallback, useContext, useRef, useState} from 'react';
+import {PressResponder, useHover} from '@react-aria/interactions';
+import React, {createContext, ForwardedRef, forwardRef, ReactElement, ReactNode, RefObject, useCallback, useContext, useRef, useState} from 'react';
+import {RootMenuTriggerState, UNSTABLE_useSubmenuTriggerState} from '@react-stately/menu';
 import {Separator, SeparatorContext} from './Separator';
 import {TextContext} from './Text';
+import {UNSTABLE_useSubmenuTrigger} from '@react-aria/menu';
 
 export const MenuContext = createContext<ContextValue<MenuProps<any>, HTMLDivElement>>(null);
 export const MenuStateContext = createContext<TreeState<unknown> | null>(null);
+export const RootMenuTriggerStateContext = createContext<RootMenuTriggerState | null>(null);
+export const SubmenuContext = createContext<{popoverContainerRef: RefObject<Element>} | null>(null);
```

Comment:
Do we need to create a separate SubmenuContext for this, or could we have Menu provide PopoverContext with UNSTABLE_portalContainer directly?

Reply:
That should work. We'll get a reading refs during render lint warning, but I guess we can ignore.